src/utils/components/CloudinitModal/utils/cloudinit-utils.ts

```diff
@@ -100,10 +106,54 @@ export const createDefaultCloudInitYAML = () =>
     true,
   );
 
+export const updateVMRHELSubscription = (
+  vm: V1VirtualMachine,
+  subscriptionData: RHELAutomaticSubscriptionData,
+): V1VirtualMachine => {
+  const cloudInitVol = getCloudInitVolume(vm);
+  const restVolumes = getVolumes(vm).filter((vol) => vol?.name !== cloudInitVol?.name);
+  const cloudInitVolData = getCloudInitData(cloudInitVol);
+
+  const userDataObject = convertYAMLUserDataObject(cloudInitVolData?.userData);
+
+  const { activationKey, organizationID } = subscriptionData || {};
+
+  const updatedUserDataObject = produce(userDataObject, (draftUserDataObject) => {
+    if (!isEmpty(organizationID) && !isEmpty(activationKey)) {
```

Comment:
i think this check should be much higher-- if its false it can avoid all this convert and functions below